llvm/test/CodeGen/X86/fold_bitcast_md_range.ll

```diff
@@ -0,0 +1,31 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc -mtriple=x86_64-apple-macosx10.12.0 -O0 < %s | FileCheck %s
+
+; Ensure that when a bitcast is folded into a load, range metadata is invalidated
+; if it does not match the new type.
+
+target datalayout = "e-m:o-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128"
+
+define void @fold_bitcast_range_metadata() {
+; CHECK-LABEL: fold_bitcast_range_metadata:
+; CHECK:       ## %bb.0: ## %start
+; CHECK-NEXT:    movaps 0, %xmm0
+; CHECK-NEXT:    movaps {{.*#+}} xmm1 = [1,0,0,0]
+; CHECK-NEXT:    pcmpeqb %xmm1, %xmm0
+; CHECK-NEXT:    pmovmskb %xmm0, %eax
+; CHECK-NEXT:    subl $65535, %eax ## imm = 0xFFFF
+; CHECK-NEXT:    jne LBB0_1
+; CHECK-NEXT:    jmp LBB0_1
+; CHECK-NEXT:  LBB0_1: ## %bb1
+; CHECK-NEXT:    retq
+start:
+  %0 = load i128, ptr null, align 16, !range !0, !noundef !1
+  %1 = icmp eq i128 %0, 1
```

Comment:
Use named values in tests, and should replace the load from null with a load from a real value. Is the control flow really necessary to reproduce?

Reply:
Updated to use named values and a non-null ptr.  Control flow is not necessary, but icmp is.